bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/handler/SunSynkInverterHandler.java

```diff
@@ -0,0 +1,416 @@
+                    this.tempInverterChargeSettings.setIntervalBatteryCapacity(Integer.valueOf(command.toString()), 6);
+                    break;
+                // Battery charging power limit
+                case CHANNEL_BATTERY_INTERVAL_1_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            1);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_2_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            2);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_3_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            3);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_4_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            4);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_5_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            5);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_6_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            6);
+                    break;
+                // Inverter control
+                case CHANNEL_INVERTER_CONTROL_TIMER:
+                    this.tempInverterChargeSettings.setPeakAndValley(Integer.valueOf(command.toString()));
+                    break;
+                case CHANNEL_INVERTER_CONTROL_ENERGY_PATTERN:
+                    this.tempInverterChargeSettings.setEnergyMode(Integer.valueOf(command.toString()));
+                    break;
+                case CHANNEL_INVERTER_CONTROL_WORK_MODE:
+                    this.tempInverterChargeSettings.setSysWorkMode(Integer.valueOf(command.toString()));
+                    break;
+            }
+
+            this.batterySettingsUpdated = true; // true = update battery settings from API at next interval
+        }
+    }
+
+    private void sendSettingsCommandToInverter() {
+        logger.debug("Ok - will handle command for CHANNEL_BATTERY_INTERVAL_1_GRID_CHARGE");
+        try {
+            inverter.sendSettings(this.tempInverterChargeSettings);
+        } catch (SunSynkSendCommandException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Could not send command to inverter " + config.getAlias() + ". Authentication Failure !");
+            return;
+        }
+        logger.debug("Sent command: to inverter {}.", config.getAlias());
```

Comment:
I would also suggest to move this logger inside `inverter.sendSettings()`
```suggestion
        logger.debug("Sens command: to inverter {}.", config.getAlias());
```

Reply:
moved to inverter.sendSettings()